BDArmory/Weapons/ModuleWeapon.cs

```diff
@@ -5668,9 +5671,9 @@ IEnumerator ReloadRoutine()
         {
             guiStatusString = "Reloading";
             hasCharged = false;
-            float timeGap = 60 / roundsPerMinute * TimeWarp.CurrentRate
+            float timeGap = 60 / roundsPerMinute * TimeWarp.CurrentRate;
             float netReloadTime = ReloadTime - timeGap - (postFireDecharge && ChargeTime > 0 ? ChargeTime : 0);
-            yield return new WaitForSecondsFixed(Mathf.Min(timeGap, fireState[0].length / fireAnimSpeed)); //wait for fire anim to finish.
+            yield return new WaitForSecondsFixed(Mathf.Min(timeGap, hasFireAnimation ? fireState[0].length / fireAnimSpeed : 0)); //wait for fire anim to finish.
```

Comment:
`WaitForSecondsFixed` always yields for at least 1 frame (since it effectively becomes `yield return null`). I'll switch this to
```csharp
if (hasFireAnimation) yield return new WaitForSecondsFixed(Mathf.Min(timeGap, fireState[0].length / fireAnimSpeed)); //wait for fire anim to finish.
```
instead before merging.